### PR TITLE
1430 validation of preflabel

### PR DESCRIPTION
**Pull request by psiotwo**

### CODE REVIEW

File: src/component/term/TermMetadataEdit.tsx [hunk 1]
```diff
@@ -58,8 +61,15 @@ export class TermMetadataEdit extends React.Component<TermMetadataEditProps, Ter
     public onPrefLabelChange = (prefLabel : string) => {
         const update = {};
         update[this.props.language] = prefLabel;
-        this.setState({labelExists: false, label: Object.assign({}, this.state.label, update)});
-        if (prefLabel.toLowerCase() === getLocalized(this.props.term.label).toLowerCase()) {
+        const updateLabelExists = {};
+        updateLabelExists[this.props.language] = false;
+        this.setState({
+            label: Object.assign({}, this.state.label, update),
+            labelExist: Object.assign({}, this.state.labelExist, updateLabelExists)
```
Comment (ledsoft) [inline src/component/term/TermMetadataEdit.tsx]: Why is the `labelExist` attribute initialized as Map, when it is on the first label edit replaced by a plain object?

**Review by ledsoft — COMMENTED** [review]
This does not resolve the same issue in the term creation form (yes, we have two components which are to a large extent duplicated).

**Comment by psiotwo** [comment]
> This does not resolve the same issue in the term creation form (yes, we have two components which are to a large extent duplicated).

Thanks for reminder. I tried to refactor to cover all cases.

File: src/component/term/TermMetadataCreateForm.tsx [hunk 2]
```diff
@@ -47,20 +46,13 @@ export class TermMetadataCreateForm extends React.Component<TermMetadataCreateFo
...
-    }
-
     private onLabelChange = (e: React.ChangeEvent<HTMLInputElement>) => {
         this.onPrefLabelChange(e.currentTarget.value);
     };
 
     private onPrefLabelChange = (label : string) => {
         const change = Object.assign({}, this.props.termData.label);
         change[this.props.language] = label;
-        this.props.onChange({label: change});
```
Comment (ledsoft) [inline src/component/term/TermMetadataCreateForm.tsx:63]: I'm sorry for keeping this blocked, but:
- Why do you create a `change` object, which is not used anywhere besides the assignment
- And then use the `checkLabelUniqueness` method, which IMHO should be a validation-only function, for the side-effect of updating the parent state via the onChange method?

**Review by ledsoft — COMMENTED** [review]
See the inline comment.

**Review by ledsoft — APPROVED** [review]
LGTM

**Pull request merged**